### ISSUE TITLE
Support "Pointer Cancellation" (W3C 2.5.2)  for `useTabList` or `useTab`

### ISSUE THREAD

**Issue by beamery-tomht** [adobe/react-spectrum#4336]
# 🙋 Feature Request

I'd like to support click cancellation for the Tabs component

Spec reference: https://www.w3.org/WAI/WCAG21/Understanding/pointer-cancellation.html

## 🤔 Expected Behavior

An option could be supplied to the `useTabList` to such as either:
- `shouldCancelOnPointerExit: boolean`
- or, `shouldSelectOnPressUp: boolean`

That allows consumers to control the press behaviour

Then a user will only confirm the selection on a `pointerup` event over the tab item that a `pointerdown` event is in progress for

## 😯 Current Behavior

The Tab is selected on the `pointerdown` event and so can't be cancelled by a user, failing criteria 2.5.2

## 💁 Possible Solution

Make the existing mechanisms of `usePress` available to the `useTabList` or `useTab`

## 🔦 Context

We're in the process of an accessibility audit and this has been flagged by our auditors on the tab component as failing compliance.

I've tried a workaround to wrap the event handlers returned from `useTab` into a `usePress` but this has been unsuccessful as the event types don't map (or at least I gave up after some hours of trying to get them to map)

## 💻 Examples

NA

## 🧢 Your Company/Team

Beamery LTD, a candidate-management-system

## 🎁 Tracking Ticket (optional)

NA, our tickets aren't public

**Comment by beamery-tomht** [comment]
That's right, our use case has a destructive action as a fresh state is fetched for each tab change

I've created this PR ready for review https://github.com/adobe/react-spectrum/pull/4342
